src/System.Management.Automation/engine/SessionStateProviderAPIs.cs

```diff
@@ -1510,285 +1452,6 @@ private void NewProviderEntry(ProviderInfo provider)
-            {
-                throw;
-            }
-            catch (ActionPreferenceStopException)
-            {
-                throw;
-            }
-            catch (Exception e) // Catch-all OK, 3rd party callout
-            {
-                // NTRAID#Windows OS Bugs-1009281-2004/02/11-JeffJon
-                this.ExecutionContext.ReportEngineStartupError(e);
-            }
-        }
-
-        private string GetProviderName(ProviderConfigurationEntry entry)
-        {
-            string name = entry.Name;
-            if (entry.PSSnapIn != null)
-            {
-                name =
-                    string.Format(
-                        System.Globalization.CultureInfo.InvariantCulture,
-                        "{0}\\{1}",
-                        entry.PSSnapIn.Name,
-                        entry.Name);
-            }
-            return name;
-        }
-
-        /// <summary>
-        /// Removes the provider of the given name.
-        /// </summary>
-        ///
-        /// <param name="providerName">
-        /// The name of the provider to remove.
-        /// </param>
-        ///
-        /// <param name="force">
-        /// Determines if the provider should be removed forcefully even if there were
-        /// drives present or errors.
-        /// </param>
-        ///
-        /// <param name="context">
-        /// The context under which the command is being run.
-        /// </param>
-        ///
-        /// <error cref="ArgumentNullException">
-        /// If <paramref name="providerName"/> is null.
-        /// </error>
-        ///
-        /// <error cref="SessionStateException">
-        /// There are still drives associated with this provider,
-        /// and the "force" option was not specified.
-        /// </error>
-        ///
-        /// <error cref="ProviderNotFoundException">
-        /// A provider with name <paramref name="providerName"/> could not be found.
-        /// </error>
-        ///
-        /// <error>
-        /// If a provider throws an exception it gets written to the <paramref name="context"/>.
-        /// </error>
-        ///
-        /// <exception cref="ArgumentException">
-        /// If <paramref name="providerName"/> is null or empty.
-        /// </exception>
-        ///
-        /// <exception cref="ArgumentNullException">
-        /// If <paramref name="context"/> is null.
-        /// </exception>
-        ///
-        /// <remarks>
-        /// All drives associated with the provider must be removed before the provider
-        /// can be removed. Call SessionState.GetDrivesForProvider() to determine if there
-        /// are any drives associated with the provider. A SessionStateException
-        /// will be written to the context if any such drives do exist.
-        /// </remarks>
-        ///
-        internal void RemoveProvider(
-            string providerName,
-            bool force,
-            CmdletProviderContext context)
```

Comment:
Same question here. This method doesn't look like solely related to RunspaceConfiguration.

Reply:
Putting back

Reply:
These following two methods seem OK to be removed:
```
private void RemoveProvider(ProviderConfigurationEntry entry)
private string GetProviderName(ProviderConfigurationEntry entry)
```

Reply:
Will remove.